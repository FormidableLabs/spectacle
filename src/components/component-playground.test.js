import React from "react";
import { render, shallow } from "enzyme";
import { renderToJson } from "enzyme-to-json";
import ComponentPlayground from "./component-playground";

describe("<ComponentPlayground />", () => {
  test("Should render the dark theme correctly", () => {
    const wrapper = render(
      <ComponentPlayground
        theme="dark"
      />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  test("Should render the light theme correctly", () => {
    const wrapper = render(
      <ComponentPlayground
        theme="light"
      />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  test("Should render with a custom background color", () => {
    const wrapper = render(
      <ComponentPlayground
        theme="light"
        previewBackgroundColor="#ff0"
      />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  test("Should render with a custom code block", () => {
    const code = `
      const Button = ({ title }) => (<button type="button">{ title }</button>);
      render(<Button title="My Button" />, mountNode);
    `;
    const wrapper = render(
      <ComponentPlayground
        theme="light"
        code={code}
        previewBackgroundColor="#ff0"
      />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  test("Should render custom scoped components", () => {
    const NewComponent = () => (
      <div><h1>Hi!</h1></div>
    );
    const wrapper = shallow(
      <ComponentPlayground
        scope={{ NewComponent }}
      />);

    const scope = wrapper.find("ReactPlayground").prop("scope");
    expect(scope.NewComponent).toEqual(NewComponent);
  });
});
