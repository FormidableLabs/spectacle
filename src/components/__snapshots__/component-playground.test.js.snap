// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentPlayground /> Should render the dark theme correctly 1`] = `
<div
  class="theme-dark jELORc"
>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          /**
           * Sample React Component 
           * Output domContainerNode is 'mountNode'
           */
          
          const styles = {
            heading: {
              fontSize: "2.25rem",
              fontWeight: "bold"
            },
            copy: {
              fontSize: "1.5rem"
            }
          }
          
          const HelloWorld = ({ name }) =&gt; (
            &lt;div&gt;
              &lt;h1 style={styles.heading}&gt;
                Create Live Code Examples in {name}!
              &lt;/h1&gt;
              &lt;p style={styles.copy}&gt;
                Supports Light and Dark Syntax Themes
              &lt;/p&gt;
            &lt;/div&gt;
          )
          
          render(
            &lt;HelloWorld name="Spectacle" /&gt;, 
            mountNode)
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ComponentPlayground /> Should render the light theme correctly 1`] = `
<div
  class="theme-light jELORc"
>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          /**
           * Sample React Component 
           * Output domContainerNode is 'mountNode'
           */
          
          const styles = {
            heading: {
              fontSize: "2.25rem",
              fontWeight: "bold"
            },
            copy: {
              fontSize: "1.5rem"
            }
          }
          
          const HelloWorld = ({ name }) =&gt; (
            &lt;div&gt;
              &lt;h1 style={styles.heading}&gt;
                Create Live Code Examples in {name}!
              &lt;/h1&gt;
              &lt;p style={styles.copy}&gt;
                Supports Light and Dark Syntax Themes
              &lt;/p&gt;
            &lt;/div&gt;
          )
          
          render(
            &lt;HelloWorld name="Spectacle" /&gt;, 
            mountNode)
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ComponentPlayground /> Should render with a custom background color 1`] = `
<div
  class="theme-light TlFwj"
>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          /**
           * Sample React Component 
           * Output domContainerNode is 'mountNode'
           */
          
          const styles = {
            heading: {
              fontSize: "2.25rem",
              fontWeight: "bold"
            },
            copy: {
              fontSize: "1.5rem"
            }
          }
          
          const HelloWorld = ({ name }) =&gt; (
            &lt;div&gt;
              &lt;h1 style={styles.heading}&gt;
                Create Live Code Examples in {name}!
              &lt;/h1&gt;
              &lt;p style={styles.copy}&gt;
                Supports Light and Dark Syntax Themes
              &lt;/p&gt;
            &lt;/div&gt;
          )
          
          render(
            &lt;HelloWorld name="Spectacle" /&gt;, 
            mountNode)
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ComponentPlayground /> Should render with a custom code block 1`] = `
<div
  class="theme-light TlFwj"
>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          const Button = ({ title }) =&gt; (&lt;button type="button"&gt;{ title }&lt;/button&gt;);
                render(&lt;Button title="My Button" /&gt;, mountNode);
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
</div>
`;
